pkg/policy/input_resolver.go

```diff
@@ -39,8 +40,14 @@ func (r *inputResolver) resolve(ctx context.Context, query ResourcesQuery) (Reso
 	ret := ResourcesResult{ScopeFound: true}
 	if resources, ok := r.input.Resources[query.ResourceType]; ok {
 		ret.ScopeFound = true
-		for _, resource := range resources {
-			ret.Resources = append(ret.Resources, resource)
+		keys := make([]string, 0, len(resources))
+		ret.Resources = make([]models.ResourceState, 0, len(resources))
```

Comment:
Minor nit: I'd move this line down to before the `for _, k := range keys` loop to keep the code that uses it closer together.
"up here" it's confusing as it's not used for the next "operation" (creating & sorting the keys-slice)